runtime/gc_base/StringTable.cpp

```diff
@@ -532,21 +532,35 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 	MM_StringTable *stringTable = MM_GCExtensions::getExtensions(vm->omrVM)->getStringTable();
 	j9object_t result = NULL;
 	j9object_t charArray = NULL;
-	UDATA allocateFlags = J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE;
+	bool compressStrings = IS_STRING_COMPRESSION_ENABLED_VM(vm);
 	bool isUnicode = J9_ARE_ANY_BITS_SET(stringFlags, J9_STR_UNICODE);
+	bool translateSlashes = J9_ARE_ANY_BITS_SET(stringFlags, J9_STR_XLAT);
+	bool anonClassName = J9_ARE_ANY_BITS_SET(stringFlags, J9_STR_ANON_CLASS_NAME);
+	bool internString = J9_ARE_ANY_BITS_SET(stringFlags, J9_STR_INTERN);
 
+	Trc_MM_createJavaLangString_Entry(vmThread, length, data, stringFlags);
+
+	/* Determine if the string contains only ASCII (<= 0x7F) characters */
+	bool isASCII = true;
 	if (isUnicode) {
-		if (IS_STRING_COMPRESSION_ENABLED_VM(vm) && VM_VMHelpers::isUnicodeASCII((const U_16 *)data, length / 2)) {
-			stringFlags |= J9_STR_ASCII;
+		/* Translation and anon class name are not currently supported for unicode input */
+		Assert_MM_false(translateSlashes || anonClassName);
+		U_16 *unicodeData = (U_16*)data;
```

Comment:
This discards the optimization of only needing to determine isASCII when compressStrings is true.

Reply:
nm, I see you are using it below.

Reply:
Ah, but not in the isUnicode case, so my original comment stands.

Reply:
```
		if (isUnicode) {
			U_16 * unicodeData = (U_16 *) data;

			if (compressStrings && isASCII) {
				for (UDATA i = 0; i < unicodeLength; ++i) {
					J9JAVAARRAYOFBYTE_STORE(vmThread, charArray, i, (I_8)unicodeData[i]);
				}
```

Your previous code actually removed this optimization (which was hidden in the VMHelpers code).

Reply:
Hmm, I might be wrong about the actual impact - looks like isASCII is only used  for unicode if compression is on. Part of the reason for this cleanup is that the logic was so impenetrable.

I'll re-add the optimization with a comment - I suppose the worst thing that could happen with any future changes is that we take the long path for unicode, which would still yield the correct result.

Reply:
Fixed.